force-app/main/default/classes/TriggerHandler.cls

```diff
@@ -0,0 +1,283 @@
+public virtual class TriggerHandler {
+  // static map of handlername, times run() was invoked
+  private static Map<String, LoopCount> loopCountMap;
+  private static Set<String> bypassedHandlers;
+
+  // custom setting which allows to bypass triggers (based on profile)
+  private Run_All_Triggers__c triggersSetting = Run_All_Triggers__c.getInstance(
+    UserInfo.getProfileId()
+  );
+
+  // the current context of the trigger, overridable in tests
+  @TestVisible
+  private TriggerContext context;
+
+  // the current context of the trigger, overridable in tests
+  @TestVisible
+  private Boolean isTriggerExecuting;
+
+  // static initialization
+  static {
+    loopCountMap = new Map<String, LoopCount>();
+    bypassedHandlers = new Set<String>();
+  }
+
+  // constructor
+  public TriggerHandler() {
+    this.setTriggerContext();
+  }
+
+  /***************************************
+   * public instance methods
+   ***************************************/
+
+  // main method that will be called during execution
+  public void run() {
+    if (!validateRun() || triggersSetting.Value__c) {
```

Comment:
I added the management of the custom setting inside the run() method which is called by the Trigger. The declaration and initialization of this setting is on line 7. I don't know if it's the best way to handle it (within this if). I wanted to add it directly in the trigger, but I know that wouldn't be right since the trigger has to be logic-less.